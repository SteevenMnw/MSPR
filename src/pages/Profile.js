//import liraries
import React from "react";
import { View, Text, StyleSheet } from "react-native";
import AccountInfo from "../components/AccountInfo";

// create a component
class Profile extends React.Component {
  constructor() {
    super();
  }
  render() {
    return (
      <View testID="view_container" style={styles.container}>
        <Text testID="text_monCompte" style={styles.text}>
          Mon Compte
        </Text>
        {/* display component information of AccountInfo */}
        <AccountInfo />
      </View>
    );
  }
}

// define your styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#E8E8E8",
  },
  text: {
    fontSize: 40,
    paddingLeft: 15,
    paddingTop: 5,
    fontWeight: "bold",
    color: "#ba473c",
  },
});

//make this component available to the app
export default Profile;
