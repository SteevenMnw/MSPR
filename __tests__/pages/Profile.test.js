import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Profile from "../../src/pages/Profile";
import { render } from "@testing-library/react-native";
import { expect, it } from "@jest/globals";

it("render of Profile page", () => {
  const display = renderer.create(<Profile />).toJSON();
  expect(display).toMatchSnapshot();
});

//Permet de generer un test
it("renders default elements and styles", () => {
  const { getAllByText, getByTestId } = render(<Profile />);

  //Permet de verifier si le text bien(et qu'il en existe 1 de chaque)
  expect(getAllByText("Mon Compte").length).toBe(1);

  //Permet de verifier que le style correspond bien
  expect(getByTestId(/view_container/i).props.style).toMatchSnapshot(
    {
      flex: 1,
      backgroundColor: "#E8E8E8",
    },
    "test container style"
  );

  expect(getByTestId(/text_monCompte/i).props.style).toMatchSnapshot(
    {
      fontSize: 40,
      paddingLeft: 15,
      paddingTop: 5,
      fontWeight: "bold",
      color: "#ba473c",
    },
    "test textMonCompte style"
  );
});
