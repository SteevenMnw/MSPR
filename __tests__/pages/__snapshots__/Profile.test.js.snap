// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> render of Profile page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E8E8E8",
      "flex": 1,
    }
  }
  testID="view_container"
>
  <Text
    style={
      Object {
        "color": "#ba473c",
        "fontSize": 40,
        "fontWeight": "bold",
        "paddingLeft": 15,
        "paddingTop": 5,
      }
    }
    testID="text_monCompte"
  >
    Mon Compte
  </Text>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "borderColor": "#F05454",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 15,
          "padding": 15,
        }
      }
      testID="test_champView1"
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        Nom
      </Text>
      <Text
        style={
          Object {
            "marginHorizontal": 123,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "borderColor": "#F05454",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 15,
          "padding": 15,
        }
      }
      testID="test_champView2"
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        Prenom
      </Text>
      <Text
        style={
          Object {
            "marginHorizontal": 123,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "borderColor": "#F05454",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 15,
          "padding": 15,
        }
      }
      testID="test_champView3"
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        Email
      </Text>
      <Text
        style={
          Object {
            "marginHorizontal": 123,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffff",
          "borderColor": "#F05454",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 15,
          "padding": 15,
        }
      }
      testID="test_champView4"
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "paddingLeft": 10,
          }
        }
      >
        Mot de passe
      </Text>
      <Text>
        
      </Text>
      <Text />
    </View>
  </View>
</View>
`;

exports[`<Profile /> renders default elements and styles: test container style 1`] = `
Object {
  "backgroundColor": "#E8E8E8",
  "flex": 1,
}
`;

exports[`<Profile /> renders default elements and styles: test textMonCompte style 1`] = `
Object {
  "color": "#ba473c",
  "fontSize": 40,
  "fontWeight": "bold",
  "paddingLeft": 15,
  "paddingTop": 5,
}
`;
